### PR TITLE
Lasagna-master implementation.

### PR DESCRIPTION
**Pull request by ggallen**

### CODE REVIEW

File: lasagna-master/gallen/lasagna_master.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+func AddSecretIngredient(friendsList, myList []string) []string {
+    return append(myList, friendsList[len(friendsList)-1])
+}
+
+func ScaleRecipe(amounts []float64, portions int) []float64 {
+    var result []float64
+    for _, v := range amounts {
+        result = append(result, v / float64(2) * float64(portions))
```
Comment (davidmogar) [inline lasagna-master/gallen/lasagna_master.go:28]: You can extract `float64(portions) / 2.0` out of the loop.

Comment (ggallen) [reply lasagna-master/gallen/lasagna_master.go:28]: I don't understand.

Comment (davidmogar) [reply lasagna-master/gallen/lasagna_master.go:28]: Only `v` changes within the loop but for each iteration you are doing `float64(portions) / 2.0` (or `1 / float64(2) * float64(portions)` in your code) which is a constant.

Comment (ggallen) [reply lasagna-master/gallen/lasagna_master.go:28]: I am not doing `float64(portions) / 2.0` anywhere in my loop.

Comment (davidmogar) [reply lasagna-master/gallen/lasagna_master.go:28]: Yes, you are doing it here ` v / float64(2) * float64(portions`. That's literally `v` multiplied by `1 / float64(portions)` multiplied by `float64(portions)` being the last two a constant.

I explained myself like shit. Sorry for that :sweat_smile:

Comment (ggallen) [reply lasagna-master/gallen/lasagna_master.go:28]: Ah, now I get it. Thanks.

I think that would warrant a detailed comment to indicate what exactly is happening. While I now understand the math transformation you are making, it's not intuitive, at least to me.

Comment (davidmogar) [reply lasagna-master/gallen/lasagna_master.go:28]: Absolutely right. I will next time!

Comment (ggallen) [reply lasagna-master/gallen/lasagna_master.go]: Change applied. Thanks for the feedback.

File: lasagna-master/gallen/lasagna_master.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+func AddSecretIngredient(friendsList, myList []string) []string {
+    return append(myList, friendsList[len(friendsList)-1])
+}
+
+func ScaleRecipe(amounts []float64, portions int) []float64 {
+    var result []float64
+    // Simplification: v/2 * portions  ===  v * 1/2 * portions === v * portions/2
+    var factor = float64(portions)/float64(2)
```
Comment (davidmogar) [inline lasagna-master/gallen/lasagna_master.go:28]:
```suggestion
    var factor = float64(portions)/2.0
```

Comment (ggallen) [reply lasagna-master/gallen/lasagna_master.go]: Yeah, good catch.

**Pull request merged**